core/src/main/java/it/pagopa/selfcare/external_interceptor/core/SchedulerServiceImpl.java

```diff
@@ -80,28 +80,30 @@ void regenerateQueueNotifications() {
 
     private void sendSapNotifications(List<Token> tokens) {
         tokens.forEach(token -> {
-            Institution institution = msCoreConnector.getInstitutionById(token.getInstitutionId());
-            RootParent rootParent = new RootParent();
-            rootParent.setDescription(institution.getParentDescription());
-            Notification toNotify = token.getStatus().equals(RelationshipState.ACTIVE)
-                    ?notificationMapper.toNotificationToSend(institution, token, QueueEvent.ADD)
-                    :notificationMapper.toNotificationToSend(institution, token, QueueEvent.UPDATE);
-            try {
-                InstitutionProxyInfo institutionProxyInfo = partyRegistryProxyConnector.getInstitutionProxyById(institution.getExternalId());
-                institution.setIstatCode(institutionProxyInfo.getIstatCode());
-                GeographicTaxonomies geographicTaxonomies = partyRegistryProxyConnector.getExtById(institution.getIstatCode());
-                institution.setCounty(geographicTaxonomies.getProvinceAbbreviation());
-                institution.setCountry(geographicTaxonomies.getCountryAbbreviation());
-                institution.setCity(geographicTaxonomies.getDescription().replace(DESCRIPTION_TO_REPLACE_REGEX, ""));
-            } catch (ResourceNotFoundException e) {
-                log.warn("Error while searching institution {} on IPA, {} ", institution.getExternalId(), e.getMessage());
-                institution.setIstatCode(null);
-            }
-            toNotify.setInstitution(institution);
-            try {
-                kafkaSapSendService.sendOldEvents(toNotify);
-            } catch (JsonProcessingException e) {
-                throw new RuntimeException(e);
+            if(token.getStatus().equals(RelationshipState.ACTIVE)) {
```

Comment:
you might avoid accidentally nullpointer inverting the check, for example RelationshipState.ACTIVE.equals(token.getStatus())